sqlserver/datadog_checks/sqlserver/queries.py

```diff
@@ -156,3 +156,61 @@
         {'name': 'failover_cluster', 'type': 'tag'},
     ],
 }
+
+
+def get_query_file_stats(sqlserver_major_version):
+    """
+    Construct the dm_io_virtual_file_stats QueryExecutor configuration based on the SQL Server major version
+
+    :param sqlserver_major_version: SQL Server major version (i.e. 2012, 2019, ...)
+    :return: a QueryExecutor query config object
+    """
+
+    column_definitions = {
+        'size_on_disk_bytes': {'name': 'files.size_on_disk', 'type': 'gauge'},
+        'num_of_reads': {'name': 'files.reads', 'type': 'monotonic_count'},
+        'num_of_bytes_read': {'name': 'files.read_bytes', 'type': 'monotonic_count'},
+        'io_stall_read_ms': {'name': 'files.read_io_stall', 'type': 'monotonic_count'},
+        'io_stall_queued_read_ms': {'name': 'files.read_io_stall_queued', 'type': 'monotonic_count'},
+        'num_of_writes': {'name': 'files.writes', 'type': 'monotonic_count'},
+        'num_of_bytes_written': {'name': 'files.written_bytes', 'type': 'monotonic_count'},
+        'io_stall_write_ms': {'name': 'files.write_io_stall', 'type': 'monotonic_count'},
+        'io_stall_queued_write_ms': {'name': 'files.write_io_stall_queued', 'type': 'monotonic_count'},
+        'io_stall': {'name': 'files.io_stall', 'type': 'monotonic_count'},
+    }
+
+    if sqlserver_major_version <= 2012:
+        column_definitions.pop("io_stall_queued_read_ms")
+        column_definitions.pop("io_stall_queued_write_ms")
+
+    # sort columns to ensure a static column order
+    sql_columns = []
+    metric_columns = []
+    for column in sorted(column_definitions.keys()):
+        sql_columns.append("fs.{} AS {}".format(column, column))
```

Comment:
nit: the `AS` is no longer necessary since column order is the only thing that matters with QueryExecutor.

Reply:
oh good point, removed